docs/design/proxy-support.md

```diff
@@ -0,0 +1,222 @@
+# Supporting package managers proxies in Hermeto
+
+## Background
+
+It might be desirable to pull dependencies through a proxy for a variety of
+reasons (builds reproducibility, additional verification, fetch speedup,
+etc). This means that Hermeto has to support proxying packages for
+individual package managers.
+
+## Proxy usage overview
+
+Most supported package managers (with a notable exception of Yarn v1) make
+use of proxy URL and support standard mechanisms for authentication.  Proxy
+usage is transparent to users as long as they provide correct proxy
+address and authentication credentials. Given that a proxy is correctly
+configured a user would just receive their dependencies as usual with an
+underlying package manager handling everything seamlessly.  Thus a way to
+consume proxy URL and credentials from the environment must be added to
+Hermeto.
+
+A proxy might require authentication or not.  A proxy might exist for just
```

Comment:
nitpick:
`s/A proxy might exist for just some package managers and do not exist for others/Note that a proxy might only be set up for a subset of the supported package managers.`